src/components/gamespec/Board.js

```diff
@@ -57,18 +92,80 @@ class Board extends React.Component {
   }
 
   handleDragEnd = (index) => {
+
+    let pieces = Object.assign([],this.props.getPieces())
     let items = this.props.getItems();
-    let item = items[index];
+    let copy = Object.assign([], items);
+    
     let position = this.refs["canvasImage" + index].refs.image.getAbsolutePosition();
-    item.offset.x = position.x;
-    item.offset.y = position.y;
-    items[index] = item;
-    this.props.setItems(items);
+    let indice = [];
+    let flag = true;
+    let offset = {
+      x: position.x,
+      y: position.y
+    }
+
+      for(let j = 0; j < copy.length && flag ; j++){
+        let off = copy[j].offset;
+        let x = off.x - offset.x
+        let y = off.y - offset.y
+        if(x * x < Math.pow(copy[j].image.width / 4, 2) && y * y < Math.pow(copy[j].image.height / 4, 2)){
+          if(copy[index].indice[0] === copy[j].indice[0]){
+            copy[index].offset.x = position.x
+            copy[index].offset.y = position.y
+          } else {
+            console.log("pieces:" + pieces)
+            console.log("copy:" + copy)
+            let idx = copy[j].indice[0];
+            pieces[idx].pieceIdx.push(index)
+            indice.push(idx);
+            indice.push(pieces[idx].pieceIdx.length - 1);
+            copy[index].offset.x = position.x
```

Comment:
Instead of setting a lot of things in multiple if else, set the variable named index and in the end after all of if else set the copy's index variable's information finally. DRY.

Reply:
can you explain it clearer?

Reply:
I mean you are doing same thing like `copy[index].offset.x = position.x` in all of the if else. You can only set it once.

Reply:
emm, cool